### PR TITLE
Fix language selector - Closes #56

### LINKED ISSUE #56: Fix country/language selector

### PR DESCRIPTION
**Pull request by peter-gerhard**
@lauraluiz @schleichardt please review

### CODE REVIEW

File: app/controllers/HomeController.java [hunk 1]
```diff
@@ -33,6 +36,12 @@ public HomeController(final ControllerDependency controllerDependency) {
         return F.Promise.pure(getResult(userContext(languageTag)));
     }
 
+    public F.Promise<Result> changeLanguage() {
+        final Form<LocationFormData> filledForm = bindForm();
```
Comment (schleichardt) [inline app/controllers/HomeController.java]: The form is never checked if the data is valid.

File: app/controllers/HomeController.java [hunk 2]
```diff
@@ -52,4 +61,25 @@ private Result getResult(final UserContext userContext) {
 
         return ok(templateService().renderToHtml("home", pageData(userContext, content)));
     }
+
+    private Form<LocationFormData> bindForm() {
+        return Form.form(LocationFormData.class).bindFromRequest();
+    }
+
+    public static class LocationFormData extends Base {
+        @Constraints.Required
```
Comment (schleichardt) [inline app/controllers/HomeController.java]: You can use `Constraints.Pattern` to allow only certain languages.

**Comment by schleichardt** [comment]
:+1:

File: app/controllers/HomeController.java (hunk 1 above)
Comment (schleichardt) [inline app/controllers/HomeController.java]: But maybe fur the dmexco it is enough

Comment (peter-gerhard) [inline app/controllers/HomeController.java]: What would you expect to happen if it is wrong? I would redirect to /en/home
if you put whatever wrong thing in there you will be redirected anyway

Comment (schleichardt) [inline app/controllers/HomeController.java]: It is just experience, then you think nothing can go wrong sb. will prove you wrong. But as I said, it is okay for the dmexco.

File: conf/messages.en [hunk 3]
```diff
@@ -42,10 +42,15 @@ form.title.label = Title
...
 header.countries = Location
 header.help = HELP
+header.language = Language
 header.stores = STORES
 
 home.pageName = Home
 
+language.german = German
+language.english = English
+language.italian = Italian
```
Comment (lauraluiz) [inline conf/messages.en]: Ignore it for now, but actually I think language names have to be translated to the language (i.e. Deutsch, English, Italiano). That is the only way you can find your language when you can't understand the current one.

File: conf/templates/common/header.hbs [hunk 4]
```diff
@@ -30,18 +30,14 @@
...
-                                </select>
-                                <span class="location-dropdown-label">Country</span>
-                                <select id="{{header.location.countrySelectId}}" class="select location-select" name="country">
-                                    {{#each header.location.country}}
-                                        <option id="{{countryOptionId}}" value="{{value}}">{{text}}</option>
-                                    {{/each}}
-                                </select>
+                                <form id="location-form" action="/language" method="POST" onchange="this.submit()">
+                                    <span class="location-dropdown-label">{{header.languagesText}}</span>
+                                    <select id="dasdasdasdasd" class="select location-select" name="languageTag">
```
Comment (lauraluiz) [inline conf/templates/common/header.hbs]: Totally unique :+1:  :D

**Comment by lauraluiz** [comment]
So currently you move to the home page when changing the language? If yes, you can merge, but please create an issue to improve it by staying in the same page. For showing the shop it is an important feature IMO.

**Pull request merged**